typify-impl/src/convert.rs

```diff
@@ -999,13 +999,46 @@ impl TypeSpace {
             Name::Unknown => todo!(),
         };
 
-        let (type_entry, metadata) = self.convert_schema_object(inner_type_name, &type_schema)?;
+        let (mut type_entry, metadata) =
+            self.convert_schema_object(inner_type_name, &type_schema)?;
 
         // Make sure all the values are valid.
         enum_values
             .iter()
             .try_for_each(|value| type_entry.validate_value(self, value).map(|_| ()))?;
 
+        // If the inner type is a also a NewType than we need to additionally derive `PartialEq`.
+        // If we do not, then we will generate a compilation error by trying to generate:
+        //
+        // ```
+        // impl TryFrom<InnerType> for OuterType {
+        //   type Error = &'static str;
+        //   fn try_from(value: InnerType) -> Result<Self, Self::Error> {
+        //     if ![
+        //         InnerType("a".to_string()),
+        //         InnerType("b".to_string()),
+        //         InnerType("c".to_string()),
+        //     ]
+        //     .contains(&value)
+        //     {
+        //         Err("invalid value")
+        //     } else {
+        //         Ok(Self(value))
+        //     }
+        //   }
+        // }
+        // ```
+        // This will fail due to InnerType not implementing `PartialEq`
+        // TODO: What other conditions will cause this to fail?
+        if let TypeEntry {
+            details: TypeEntryDetails::Newtype(_),
+            derives,
+            ..
+        } = &mut type_entry
+        {
+            derives.insert("PartialEq".to_string());
```

Comment:
I'm confused: I thought we needed the inner type to be `PartialEq`, but it looks like this is for the outer type?

Reply:
I believe `type_entry` here is the inner type that gets created above from `inner_type_name`, with `newtype_entry` below being the outer type. (at least from my understanding during some test runs). Is that not correct?

Reply:
Sounds good. That was my other thought for implementing this. I'll move it to `output_newtype`